lib/datadog/appsec/waf.rb

```diff
@@ -482,6 +479,33 @@ class Handle
 
         attr_reader :ruleset_info
 
+        def self.update(handle, data)
+          data_obj = Datadog::AppSec::WAF.ruby_to_object(data, coerce: false)
+          ruleset_info = LibDDWAF::RuleSetInfo.new
+          new_handle = Datadog::AppSec::WAF::LibDDWAF.ddwaf_update(handle.handle_obj, data_obj, ruleset_info)
+
+          return if new_handle.null?
+
+          info = {
+            loaded: ruleset_info[:loaded],
+            failed: ruleset_info[:failed],
+            errors: WAF.object_to_ruby(ruleset_info[:errors]),
+            version: ruleset_info[:version],
+          }
+          initialize_from_handle(new_handle, info)
+        ensure
+          Datadog::AppSec::WAF::LibDDWAF.ddwaf_object_free(data_obj) if data_obj
+          Datadog::AppSec::WAF::LibDDWAF.ddwaf_ruleset_info_free(ruleset_info) if ruleset_info
+        end
+
+        def self.initialize_from_handle(handle_object, info)
+          obj = allocate
+          obj.instance_variable_set(:@handle_obj, handle_object)
+          obj.instance_variable_set(:@ruleset_info, info)
+          obj
+        end
```

Comment:
This seems to exist to go around `new`+`initialize`, correct?

I feel like we should be able to dispense from that by having `initialize` accept a `from:` keyword argument.